lib/internal/test_runner/mock.js

```diff
@@ -157,16 +161,30 @@ class MockTracker {
         'options.setter', setter, "cannot be used with 'options.getter'"
       );
     }
+    function getOriginalObject(descriptor) {
+      let original;
+
+      if (getter) {
+        original = descriptor?.get;
+      } else if (setter) {
+        original = descriptor?.set;
+      } else {
+        original = descriptor?.value;
+      }
+
+      return original;
+    }
+    let descriptor = ObjectGetOwnPropertyDescriptor(object, methodName);
+    let original = getOriginalObject(descriptor);
 
-    const descriptor = ObjectGetOwnPropertyDescriptor(object, methodName);
-    let original;
+    // classes instances
+    if (typeof original !== 'function') {
+      descriptor = ObjectGetOwnPropertyDescriptor(
+        ObjectGetPrototypeOf(object),
```

Comment:
What if the object is a null-prototype one? This would throw.

Reply:
if the object is null the validation `validateStringOrSymbol(methodName, 'methodName');`  gets it before this line and throws `The "methodName" argument must be one of type string or symbol. Received an instance of Object`